### ISSUE TITLE
Create a public API to remove an attribute from `attributes` and `attribute_overrides`

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor#1737]
In Asciidoctor.js we show how to ["unset/delete document attribute"](https://github.com/asciidoctor/asciidoctor.js/blob/master/manual.adoc#how-to-unsetdelete-document-attribute).
With the new Asciidoctor.js, the code will look like:

```javascript
var content = '== Title';
var doc = Opal.Asciidoctor.$load(content);

// Asciidoctor Ruby API via Asciidoctor.js API (without $ sign)
doc.set_attribute('data-uri', 'true');
console.log(doc.attr('data-uri')); // prints 'true'

// Ruby API via Opal (with $ sign)
doc.attributes.$delete('data-uri');
doc.attribute_overrides.$delete('data-uri');
```

I think the use of `$` or not will be confusing.
A public API to unset/delete document attribute can solve this issue and improve readability:

```javascript
var content = '== Title';
var doc = Opal.Asciidoctor.$load(content);

doc.set_attribute('data-uri', 'true');
doc.unset_attribute('data-uri');
```

I don't know if we should always delete attribute from both `attributes` and `attribute_overrides` ?
@mojavelinux Wdyt ?

**Comment by mojavelinux** [comment]
There is a method to delete an attribute on Document, `delete_attribute`. The problem is that this method cannot be used to delete a locked attribute (which is an attribute that has been set from the API). So it's not very useful.

The `set_attribute` and `delete_attribute` methods on Document are not well designed. Technically, these are internal APIs, but even that's fuzzy. And `set_attribute` seems redundant next to `set_attr` attribute on Node. There's no equivalent `unset_attr` (or `delete_attr`) on Node.

Perhaps it's time to address this problem.

I like the idea of "unset" as a complement to "set". Though, the terms "set" and "delete" were chosen since they match the terminology of Ruby Hash.

The first thing we should do is add `unset_attr` to Node (perhaps aliased to `remove_attr` or `delete_attr`). That method will work much like `set_attr` in reverse.

The second thing we need to do is figure out how to have `set_attr` and `unset_attr` on Document honor or update the attribute_overrides keys. Perhaps a third argument with options, such as `lock: true`, which locks the attribute so the document can't manipulate it.

In summary, I think we need to step back and design these methods correctly to handle the various use cases. Otherwise, we'll just be patching onto a flimsy foundation.

**Comment by mojavelinux** [comment]
To help this move forward, I'm happy if we focus just on the addition of `delete_attr` (or, if we decide, `remove_attr` or `unset_attr`).

I think we should mark `set_attribute` and `delete_attribute` as internal until/if we figure out how to roll that functionality into `set_attr` and `delete_attr`, respectively).

**Comment by Mogztter** [comment]
> Otherwise, we'll just be patching onto a flimsy foundation.

:+1:
I'm a bit confused with `attributes` and `attribute_overrides`... :pensive: 

> The first thing we should do is add unset_attr to Node (perhaps aliased to remove_attr or delete_attr). That method will work much like set_attr in reverse.

Ok, I will create a `removeAttr` method in the Asciidoctor.js API.
Do I have to call `delete` method on both `doc.attributes` and `doc.attribute_overrides` ?
